From 1ae0022e47a3aedfbe46720ac055b2a0dc2e521d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Pasi=20Sj=C3=B6holm?= <pasi.sjoholm@jollamobile.com>
Date: Tue, 24 Mar 2015 13:38:04 +0200
Subject: [PATCH 18/22] network: Do not try to set gateway from DHCPv6

Setting a gateway is handled in the kernel and needs not to be set
from ConnMan when DHCPv6 is used.

This fixes the issue with IPv6 nameserver host routes getting added
as link-local due to the gateway entry being NULL in the ipconfig
structure.

Reported by Tore Anderson.

(cherry picked from commit 0999b431204df43dce8367377c13226bb0c4a9f8)
---
 src/network.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/src/network.c b/src/network.c
index 3a3b852..f61f698 100644
--- a/src/network.c
+++ b/src/network.c
@@ -347,10 +347,6 @@ static int dhcpv6_set_addresses(struct connman_network *network)
 	if (err < 0)
 		goto err;
 
-	err = __connman_ipconfig_gateway_add(ipconfig_ipv6);
-	if (err < 0)
-		goto err;
-
 	return 0;
 
 err:
-- 
1.9.1

